spark3/src/main/java/org/apache/iceberg/spark/source/SparkTable.java

```diff
@@ -205,6 +205,9 @@ public boolean canDeleteWhere(Filter[] filters) {
     }
 
     Set<Integer> identitySourceIds = table().spec().identitySourceIds();
+    if (identitySourceIds.isEmpty()) {
+      return true;
+    }
```

Comment:
I am not convinced that this change to `canDeleteWhere` is correct.
@aokolnychyi can you please comment? I know you added `canDeleteWhere` to `SupportsDelete` in Spark (it went into 3.1), and iiuc, some change to rewrite the query if `canDeleteWhere` returns false is needed on the Spark side but was left as to-be-implemented-later?
This change here causes TestDeleteFrom to pass (it fails with Spark 3.1 otherwise), but in my local testing, when I make this change, TestCopyOnWriteDelete hangs (commits keep failing and retrying indefinitely).